R/importSNV.TCGA.R

```diff
@@ -0,0 +1,148 @@
+#' @export
+importSNV.TCGA <- function(  fMAF,
+                             rSNV,
+                             silent = FALSE,
+                             writeLog = TRUE,
+                             na.rm = TRUE) {
+
+    # ==== PARAMETERS ==========================================================
+
+    # NL <- .PlatformLineBreak()
+    # # Read header and one contents line
+    # tmp <- readLines(fName, n = 2)
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fMAF,         like = rep("FILE_E", length(fMAF)),
+                                                            checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(na.rm,        like = logical(1), checkSize = TRUE))
+
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+
+    #  Validate the vector of MAF file names parameter by checking
+    #  extension (.maf, .txt, .tsv. gz(?), .zip(?)) and header
+
+    #  Validate rSNV as an existing file
+    #  If not, create rSNV file as empty
+    #  If rSNV exists, load file
+    #  write to myNotes how many lines/rows did rSNV start with
+
+    myNotes <- c("Number of rows originally in rSNV - ")
+
+    # ==== READ DATA ===========================================================
+
+    # reading data will occur in multiple steps to avoid storing too much data on
+    # the disk drive and cause overload when handeling large files
+
+    # step 1: go through fMAF input vector list of files and open first file
+    read::read_delim()
+    # open columns 1,5,6,7,8,9,10,11,12,13,16,33
+    # step 2: read lines from file and process data as indicated in "EXTRACT" below
+    # typecheck and error handling
+    # step 3: save processed data to new/merge to rSNV file
+    #         merge temp data to dataframe and delete temp data
+    # step 4: close file
+    # step 5: repeat steps 1-4 for next disease MAF file avilable.
+
+    # ==== "EXTRACT" - REQUIRED DATA ========================================================
+
+    # required data from MAF files for rSNV lie in the columns:
+    # 1 HUGO SYMBOL
+    # 5 chromosome
+    # 6 start
+    # 7 end
+    # 8 strand
+    # 9 Variant_Classification
+    # 10 Variant_type
+    # 11 Reference_allele
+    # 12 Tumor_Seq_Allele1
+    # 13 Tumor_Seq_Allele2
+    # 16 Tumor_Sample_Barcode
+    # 33 Tumor_Sample_UUID
+
+    # read each sample from file and parse the columns listed above and write them
+    # to the rSNV file.
+    # write missing/dropped to log HERE
+    # incase of missing value add index of sample entry in rSNV to vector 'missingValues'
+    # when we reach the last sample close file and open new file.
+
+    # ==== "HANDLE" - MISSING VALUES =============================================
```

Comment:
You have to be more explicit about the expected function behaviour for this section: what **exactly** is a "missing value". NA? Something else too? Empty cells? Does ```readr::read_delim()``` turn empty cells into NA? Etc. If you are not 100% specific, you can't define what you will test for and how your tests should check whether the function works correctly. Not to go overboard here, I don't think we need to check for every possible thing that could go wrong - we'll get to erroneous input when we go over the test design - but the way this is written I am not sure what you mean, so as the code reviewer I need more guidance.